third_party/com_github_aws_cpp_sdk/aws_cpp_sdk.BUILD.bazel

```diff
@@ -0,0 +1,180 @@
+        "aws-cpp-sdk-core/include/aws/core/external/tinyxml2/*.h",  # TINYXML2_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/http/curl/*.h",  # HTTP_CURL_CLIENT_HEADERS
+        "aws-cpp-sdk-core/include/aws/core/utils/crypto/openssl/*.h",  # UTILS_CRYPTO_OPENSSL_HEADERS
+    ]),
+    defines = [
+        'AWS_SDK_VERSION_STRING=\\"1.7.366\\"',
+        "AWS_SDK_VERSION_MAJOR=1",
+        "AWS_SDK_VERSION_MINOR=7",
+        "AWS_SDK_VERSION_PATCH=366",
+        "ENABLE_OPENSSL_ENCRYPTION=1",
+        "ENABLE_CURL_CLIENT=1",
+        "OPENSSL_IS_BORINGSSL=1",
+    ] + select({
+        "@bazel_tools//src/conditions:windows": [
+            "PLATFORM_WINDOWS",
+            "WIN32_LEAN_AND_MEAN",
+        ],
+        "//conditions:default": [
+            "PLATFORM_LINUX",
+        ],
+    }),
+    includes = [
+        "aws-cpp-sdk-core/include",
+    ],
+    linkopts = select({
+        "@bazel_tools//src/conditions:windows": [
+            "-DEFAULTLIB:userenv.lib",
+            "-DEFAULTLIB:version.lib",
+        ],
+        "//conditions:default": [],
+    }),
+    deps = [
+        "@com_github_aws_c_event_stream//:aws-c-event-stream",
+        "@com_google_boringssl//:crypto",
+        "@com_google_boringssl//:ssl",
+        "@se_curl//:curl",
+    ],
+)
+
+cc_library(
+    name = "s3",
+    srcs = glob([
+        "aws-cpp-sdk-s3/source/*.cpp",  # AWS_S3_SOURCE
+        "aws-cpp-sdk-s3/source/model/*.cpp",  # AWS_S3_MODEL_SOURCE
+    ]),
+    hdrs = glob([
+        "aws-cpp-sdk-s3/include/aws/s3/*.h",  # AWS_S3_HEADERS
+        "aws-cpp-sdk-s3/include/aws/s3/model/*.h",  # AWS_S3_MODEL_HEADERS
+    ]),
+    includes = [
+        "aws-cpp-sdk-s3/include",
+    ],
+    deps = [
+        ":core",
+    ],
+)
+
+cc_library(
+    name = "transfer",
+    srcs = glob([
+        "aws-cpp-sdk-transfer/source/transfer/*.cpp",  # TRANSFER_SOURCE
+    ]),
+    hdrs = glob([
+        "aws-cpp-sdk-transfer/include/aws/transfer/*.h",  # TRANSFER_HEADERS
+    ]),
+    includes = [
+        "aws-cpp-sdk-transfer/include",
+    ],
+    deps = [
+        ":core",
+        ":s3",
+    ],
+)
+
+cc_library(
+    name = "kinesis",
+    srcs = glob([
+        "aws-cpp-sdk-kinesis/source/*.cpp",  # AWS_KINESIS_SOURCE
+        "aws-cpp-sdk-kinesis/source/model/*.cpp",  # AWS_KINESIS_MODEL_SOURCE
+    ]),
+    hdrs = glob([
+        "aws-cpp-sdk-kinesis/include/aws/kinesis/*.h",  # AWS_KINESIS_HEADERS
+        "aws-cpp-sdk-kinesis/include/aws/kinesis/model/*.h",  # AWS_KINESIS_MODEL_HEADERS
+    ]),
+    includes = [
+        "aws-cpp-sdk-kinesis/include",
+    ],
+    deps = [
+        ":core",
+    ],
+)
+
+genrule(
+    name = "SDKConfig_h",
+    outs = [
+        "aws-cpp-sdk-core/include/aws/core/SDKConfig.h",
+    ],
+    cmd = "\n".join([
+        "cat <<'EOF' >$@",
+        "#define USE_AWS_MEMORY_MANAGEMENT",
+        "#if defined(_MSC_VER)",
+        "#include <Windows.h>",
```

Comment:
We might want to add something like:

#ifndef WIN32_LEAN_AND_MEAN
#define WIN32_LEAN_AND_MEAN
#endif

before #include <Windows.h>